src/main.cc

```diff
@@ -2,26 +2,30 @@
 #include <memory>
 
 #include "gmsh.h"
+#include "io.h"
+#include "json.hpp"
 
 int main(int argc, char** argv) {
   try {
     if (argc != 3) {
-      std::cout << "Usage: ./mpm-point-generator /path/to/mesh_file.msh\t"
-                   "Dimension 2 or 3\n";
+      std::cout << "Usage: ./mpm-point-generator /path/to/path_file.txt\n"
+                   "Dimenssion 2 or 3\n";
       throw std::runtime_error("Incorrect number of input arguments");
     }
-
+    //! Get dimension or Tdim
     int dimension = std::atoi(argv[2]);
-    std::string filename = argv[1];
 
+    //! Get path file location including directory
+    std::string pathfile_name = argv[1];
+
+    //! Store them into an IO class
+    std::unique_ptr<IO> file(new IO(pathfile_name));
+
+    //! Main functions
     std::unique_ptr<GMSH> mesh(new GMSH());
-    mesh->get_vertices(filename);
-    mesh->output_vertices();
-    if (dimension == 2) {
-      mesh->output_stresses();
-    } else {
-      mesh->output_3d_stresses();
-    }
+    mesh->get_vertices(file->inputfilename());
+    mesh->output_vertices(file->outputfilename_point());
```

Comment:
The following two functions will pipe results to IO class, which will in turn write the results.